### ISSUE TITLE
StandardRB LSP server fails to initialize

### ISSUE THREAD

**Issue by benjamineskola** [helix-editor/helix#6506]
### Summary

Apologies if this is actually a standardrb bug, but as it worked correctly in neovim I thought this was the best place to start.

When configuring `standardrb` in LSP mode (providing diagnostics and autoformatting), it fails to initialise in Helix. Logs below.

### Reproduction Steps

`languages.toml`:

```toml
[[language]]
name = "ruby"
auto-format = true
language-server = { command = "standardrb", args = ["--lsp"] }
```

```sh
$ hx —log standardrb.log -vv foo.rb
```

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-03-31T18:05:46.913 helix_view::clipboard [DEBUG] Using pbcopy+pbpaste to interact with the system clipboard
2023-03-31T18:05:47.112 helix_vcs [ERROR] file is untracked
2023-03-31T18:05:47.112 helix_vcs [ERROR] failed to open diff base for /Users/ben/bin/foo.rb
2023-03-31T18:05:47.114 helix_view::editor [DEBUG] editor status: Loaded 1 file.
2023-03-31T18:05:47.114 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.03 (3cf03723)"},"processId":97471,"rootPath":"/Users/ben/bin","rootUri":"file:///Users/ben/bin","workspaceFolders":[{"name":"bin","uri":"file:///Users/ben/bin"}]},"id":0}
2023-03-31T18:05:47.119 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is not supported in this terminal (checked in 4.652292ms)
2023-03-31T18:05:47.119 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-03-31T18:05:47.315 helix_term::application [DEBUG] received editor event: IdleTimer
2023-03-31T18:05:48.071 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-03-31T18:05:48.225 helix_lsp::transport [INFO] <- {"id":0,"result":{"capabilities":{"textDocumentSync":{"openClose":true,"change":1},"documentFormattingProvider":true,"executeCommandProvider":true,"diagnosticProvider":true}},"jsonrpc":"2.0"}
2023-03-31T18:05:48.225 helix_lsp::transport [INFO] <- {"capabilities":{"diagnosticProvider":true,"documentFormattingProvider":true,"executeCommandProvider":true,"textDocumentSync":{"change":1,"openClose":true}}}
2023-03-31T18:05:48.225 helix_lsp [ERROR] failed to initialize language server: failed to parse: invalid type: boolean `true`, expected struct ExecuteCommandOptions
2023-03-31T18:05:48.317 helix_term::commands::typed [DEBUG] quitting...
2023-03-31T18:05:48.317 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-03-31T18:05:48.473 helix_term::commands::typed [DEBUG] quitting...
2023-03-31T18:05:48.473 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2023-03-31T18:05:48.473 helix_term::job [DEBUG] waiting on jobs...
2023-03-31T18:05:48.473 helix_term::job [DEBUG] waiting on jobs...
2023-03-31T18:05:48.473 helix_lsp::transport [INFO] Language server not initialized, delaying request
2023-03-31T18:05:51.475 helix_term::application [ERROR] Timed out waiting for language servers to shutdown
```
</details>

### Platform

macOS

### Terminal Emulator

wezterm 20230326-111934-3666303c

### Helix Version

helix 23.03 (3cf03723)

**Comment by the-mikedavis** [comment]
This is a bug in StandardRB: it's sending `"executeCommandProvider":true` in the initialization response but that field can't be a boolean in the spec: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#executeCommandOptions

**Comment by benjamineskola** [comment]
Thanks for confirming, I've reported it upstream.

**Comment by benjamineskola** [comment]
This is fixed in StandardRB 1.25.4.
